independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/DeploymentInjectingDependencyVisitor.java

```diff
@@ -111,19 +115,23 @@ private void processPlatformArtifact(DependencyNode node, Path descriptor) throw
             return;
         }
         final String value = rtProps.getProperty(BootstrapConstants.PROP_DEPLOYMENT_ARTIFACT);
+        appBuilder.handleExtensionProperties(rtProps, node.getArtifact().toString());
         if(value == null) {
             return;
         }
-        if(value != null) {
-            Artifact deploymentArtifact = toArtifact(value);
-            if(deploymentArtifact.getVersion() == null || deploymentArtifact.getVersion().isEmpty()) {
-                deploymentArtifact = deploymentArtifact.setVersion(node.getArtifact().getVersion());
-            }
-            node.setData(QUARKUS_DEPLOYMENT_ARTIFACT, deploymentArtifact);
-            runtimeNodes.add(node);
-            managedDeps.add(new Dependency(node.getArtifact(), JavaScopes.COMPILE));
-            managedDeps.add(new Dependency(deploymentArtifact, JavaScopes.COMPILE));
+        Artifact deploymentArtifact = toArtifact(value);
+        if(deploymentArtifact.getVersion() == null || deploymentArtifact.getVersion().isEmpty()) {
+            deploymentArtifact = deploymentArtifact.setVersion(node.getArtifact().getVersion());
         }
+        node.setData(QUARKUS_DEPLOYMENT_ARTIFACT, deploymentArtifact);
+        runtimeNodes.add(node);
+        managedDeps.add(new Dependency(node.getArtifact(), JavaScopes.COMPILE));
+        runtimeExtensionDeps.add(new Dependency(node.getArtifact(), JavaScopes.COMPILE));
```

Comment:
I guess it's safe to re-use `new Dependency(node.getArtifact(), JavaScopes.COMPILE)` created above.